static/app/views/dashboards/widgets/timeSeriesWidget/plottables/continuousTimeSeries.tsx

```diff
@@ -58,8 +58,21 @@ export abstract class ContinuousTimeSeries<
     this.config = config;
   }
 
+  /**
+   * Continuous time series names need to be unique to disambiguate them from other series. We use both the `yAxis` and the `groupBy` to create the name. This makes it possible to pass in two different time series with the same `yAxis` as long as they have different `groupBy` information.
+   */
   get name(): string {
-    return this.timeSeries.yAxis;
+    let name = `${this.timeSeries.yAxis}`;
+
+    if (this.timeSeries.groupBy?.length && this.timeSeries.groupBy.length > 0) {
```

Comment:
```suggestion
    if (this.timeSeries.groupBy?.length) {
```

nit, i don't think you need to explicitly check against 0 as it's falsy?